cmd/axelard/cmd/vald/start.go

```diff
@@ -320,30 +319,53 @@ func createJob(sub tmEvents.FilteredSubscriber, processor func(event tmEvents.Ev
 
 }
 
-func getStartBlock(stateStore StateStore, tmClient tmEvents.BlockHeightClient, logger log.Logger) (int64, error) {
-	startBlock, err := stateStore.GetState()
-	if err != nil {
-		logger.Error(err.Error())
-
-		return 0, nil
-	}
-
+// Wait until the node has synced with the network
+// and then return the block height to start listening to TM events from
+func waitTillNetworkSync(cfg config.ValdConfig, stateStore StateStore, tmClient tmEvents.SyncInfoClient, logger log.Logger) (int64, error) {
 	rpcCtx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
-	height, err := tmClient.LatestBlockHeight(rpcCtx)
+	syncInfo, err := tmClient.LatestSyncInfo(rpcCtx)
 	if err != nil {
 		return 0, err
 	}
 
-	// TODO: Make it configurable instead of a hardcoded 100
-	if height-startBlock > 100 {
-		logger.Info(fmt.Sprintf("block in state %d is too old and will start from the latest instead", startBlock))
+	cachedHeight, err := stateStore.GetState()
+	if err != nil {
+		logger.Info(fmt.Sprintf("failed to retrieve the cached block height, using the latest: %s", err.Error()))
+		cachedHeight = 0
+	} else {
+		logger.Info(fmt.Sprintf("retrieved cached block height %d", cachedHeight))
+		cachedHeight++ // Skip the block that might have already been executed
```

Comment:
```suggestion
		cachedHeight++ // Skip the block that has already been processed
```

Reply:
It's not guaranteed to have been processed (for e.g. if the process dies in the middle)

Reply:
but if it dies in the middle the cache should not have been updated. Unless someone changed that, the cache is written to when all block events are delivered to appropriate consumers